app/src/main/java/com/tari/android/wallet/ffi/LogFileObserver.kt

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright 2020 The Tari Project
+ *
+ * Redistribution and use in source and binary forms, with or
+ * without modification, are permitted provided that the
+ * following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of
+ * its contributors may be used to endorse or promote products
+ * derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+ * OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.tari.android.wallet.ffi
+
+import android.os.FileObserver
+import android.util.Log
+import java.io.File
+import java.io.InputStream
+
+/**
+ * Observes the log file and on a file modify, forwards the new line (if any) in
+ * the log file to the Android log.
+ *
+ * @author The Tari Development Team
+ */
+@Suppress("DEPRECATION")
+internal class LogFileObserver(logFilePath: String) : FileObserver(logFilePath) {
+
+    private val logTag = "FFI"
+    private val logFile = File(logFilePath)
+    /**
+     * Remember the last number of lines to be able to log any new lines.
+     */
+    private var lastNumberOfLines = 0
+
+    private fun logNewLines() {
+        val lineList = mutableListOf<String>()
+        if (logFile.exists()) {
+            val inputStream: InputStream = logFile.inputStream()
+            inputStream.bufferedReader()
+                .useLines { lines -> lines.forEach { lineList.add(it) } }
+        }
+        lineList.takeLast(lineList.size - lastNumberOfLines).forEach { logLine ->
+            Log.d(logTag, logLine)
```

Comment:
I think we should use `Logger` here.

Reply:
I did consider `Logger`, but I think `Logger` is better for sparse logs, i.e. the ones we send from the app code, such as:
![Screen Shot 2020-03-09 at 10 55 20](https://user-images.githubusercontent.com/2969501/76193291-baac7400-61f4-11ea-9086-0a0e5414e070.png)
Whereas the log from the wallet is pretty crowded, I think it looks better along with the other Android system logs.
![Screen Shot 2020-03-09 at 10 55 33](https://user-images.githubusercontent.com/2969501/76193352-d6b01580-61f4-11ea-998f-c3f5d227bbb3.png)
I think it would look very verbose with the pretty printing of `Logger`.

Reply:
Alright, let's use Log APIs then 👍